atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/queue/SweepQueueReader.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2017 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.sweep.queue;
+
+import java.util.Collection;
+import java.util.function.Consumer;
+
+public interface SweepQueueReader {
+
+    // Passes the next batch of writes to the given consumer. If the consumer returns without throwing,
+    // the batch is considered consumed, and future invocations of this method will consume only newer writes.
+    // On the other hand, if the consumer fails by throwing an exception, the same batch will be consumed
```

Comment:
keeping it simple for now. since the consumer is just issuing deletes, i don't see a reason why specific input would cause it to fail.

we could alter this to retry a certain amount before giving up on a segment - which is totally fine - but not sure we need this now

Reply:
Main thing here is probably looking at the perf/stability of range deletes in C*

Reply:
May need to allow success/failure on every WriteInfo here.